### PR TITLE
Sabrina/middleware unit

### PR DESCRIPTION
**Pull request by sabrina-li**
Add OCMock to mock FS and Segment classes
- unit test for getNewPayloadWithFSURL with input: screen and track payload
- create Mock SegContext as fake data and pass it through getNewPayloadWithFSURL

for [_fullStoryMiddleware context:context next:^(SEGContext * _Nullable newContext){}];
- fake SegContext that's not mock
- verify that `next` block is called to make sure our middleware does not stop the chain
- verify correct FS API is called depending on different kind of input and settings

### CODE REVIEW

**Review by nfriedly — COMMENTED** [review]
Awesome!

File: Example/Tests/FullStoryMiddlewareTest.m [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    SEGScreenPayload* expect = [[SEGScreenPayload alloc] initWithName:@"testScreen"
+                                                          properties:@{@"FSSessionURL": @"testURL"}
+                                                             context:emptyDict
+                                                        integrations:emptyDict];
+    SEGContext *context = [[[SEGContext alloc] initWithAnalytics:[SEGAnalytics sharedAnalytics]] modify:^(id<SEGMutableContext> _Nonnull ctx) {
+        ctx.eventType = SEGEventTypeScreen;
+        ctx.payload = screenPayload;
+    }];
+    
+    SEGScreenPayload* output = (SEGScreenPayload*)[_fullStoryMiddleware getNewPayloadWithFSURL:context];
```
Comment (HaroldFS) [inline Example/Tests/FullStoryMiddlewareTest.m]: I see you doing (what seems to be the same lines of code I could be wrong) in
`testFullStoryMiddleware_GetNewPayloadWithFSURL_TrackPayload_ReturnsTrackPayloadWithFSURL ` 
&
`testFullStoryMiddleware_GetNewPayloadWithFSURL_ScreenPayload_ReturnsScreenPayloadWithFSURL` 
on lines: `76-90` & `113-125`

Is it possible to abstract this out into a method below that returns a tuple of `expect` and `context` for your XCAssert tests below?

Comment (sabrina-li) [reply Example/Tests/FullStoryMiddlewareTest.m]: Great point! They are very similar (just different payload type). objC unfortunately doesn't have tuples, but group them  into some kind of util method(s) would a lot of sense!

Comment (HaroldFS) [reply Example/Tests/FullStoryMiddlewareTest.m]: Oh thats right >.< im so spoiled with swift.

**Review by HaroldFS — APPROVED** [review]
Besides that comment it LGTM, and its fine without that change.

**Comment by sabrina-li** [comment]
> Besides that comment it LGTM, and its fine without that change.

I realized that I didn't push a big chunk of change before you approved it. Sorry about that! This is ready to be reviewed again :)

**Review by HaroldFS — APPROVED** [review]
LGTM

**Pull request merged**